modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/SqlSchemaManagerImpl.java

```diff
@@ -106,6 +111,18 @@ public IgniteTable tableById(UUID id) {
         return table;
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public InternalSortedIndex indexById(UUID id, String idxName) {
+        InternalSortedIndex idx = indexManager.getIndexById(id);
+
+        if (idx == null) {
+            throw new IndexNotFoundException(idxName, id);
```

Comment:
This doesn't seem right to pass an `idxName` param only for case when index is not found. Probably, there should be one more constructor of `IndexNotFoundException` accepting the `Id` only

Reply:
refactored here.